src/signals.c

```diff
@@ -323,9 +327,8 @@ static void sig_handle_timer(int signum)
 #endif
 #ifndef BENCH_BUILD
 #if OS_TIMER
-	/* Some stuff only done every few seconds */
-	if (timer_save_interval < 4 ||
-	    ((timer_save_interval - timer_save_value) & 3) == 3) {
+	/* Some stuff only done every fourth second */
+	if (((timer_save_interval - timer_save_value) & 3) == 0) {
```

Comment:
Actually, checking for `== 1` (like you had before) might have been right, and `== 0` might be wrong now - perhaps `timer_save_interval` is at least 1, while `timer_save_value` can be 0, no? Have you tested this with either kind of check?

Reply:
`0` is correct and duly tested now. The `!OS_TIMER` side is less affected by other variables but I tested that as well.